src/ManagedServices/ManagedServices/Commands/NewAzureRmManagedServicesDefinition.cs

```diff
@@ -0,0 +1,125 @@
+    {
+        protected const string DefaultParameterSet = "Default";
+
+        [Parameter(ParameterSetName = DefaultParameterSet, Mandatory = false, HelpMessage = "The subscription scope.")]
+        [ScopeCompleter]
+        public string Scope { get; set; }
+
+        [Parameter(ParameterSetName = DefaultParameterSet, Mandatory = true, HelpMessage = "The name of the Registration Definition.")]
+        public string RegistrationDefinitionName { get; set; }
+
+        [Parameter(ParameterSetName = DefaultParameterSet, Mandatory = true, HelpMessage = "The Managedby Tenant Identifier.")]
+        [ValidatePattern("^({){0,1}[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}(}){0,1}$")]
+        public string ManagedByTenantId { get; set; }
+
+        [Parameter(ParameterSetName = DefaultParameterSet, Mandatory = true, HelpMessage = "The Managedby Principal Identifier.")]
+        [ValidatePattern("^({){0,1}[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}(}){0,1}$")]
+        public string PrincipalId { get; set; }
+
+        [Parameter(ParameterSetName = DefaultParameterSet, Mandatory = true, HelpMessage = "The Managed Service Providers's Role Identifier.")]
+        [ValidatePattern("^({){0,1}[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}(}){0,1}$")]
+        public string RoleDefinitionId { get; set; }
+
+        [Parameter(ParameterSetName = DefaultParameterSet, Mandatory = false, HelpMessage = "The description of the Registration Definition.")]
+        public string Description { get; set; }
+
+        [Parameter(ParameterSetName = DefaultParameterSet, Mandatory = false, HelpMessage = "The name of the plan.")]
+        public string PlanName { get; set; }
+
+        [Parameter(ParameterSetName = DefaultParameterSet, Mandatory = false, HelpMessage = "The name of the Publisher.")]
+        public string PlanPublisher { get; set; }
+
+        [Parameter(ParameterSetName = DefaultParameterSet, Mandatory = false, HelpMessage = "The name of the Product.")]
+        public string PlanProduct { get; set; }
+
+        [Parameter(ParameterSetName = DefaultParameterSet, Mandatory = false, HelpMessage = "The version number of the plan.")]
+        public string PlanVersion { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Run cmdlet in the background")]
+        public SwitchParameter AsJob { get; set; }
+
+        public Guid RegistrationDefinitionId { get; set; } = default(Guid);
+
+        public override void ExecuteCmdlet()
+        {
+            this.TryCatchExecuteCmdlet(() =>
+            {
+                ConfirmAction(MyInvocation.InvocationName,
+                    "Microsoft.ManagedServices/registrationDefinitions",
+                    () =>
+                    {
+                        Plan plan = null;
+                        if (!string.IsNullOrEmpty(this.PlanName) ||
```

Comment:
You want to do this if any of these are not null - don't some of them *have* to be non-null?

Also, you are better off using the IsParameterBound mechanism to determine if the user supplied this value (see: )  and using a ValidateNotNullOrEmpty attribute on the parameter

Finally, if any of these are actually required to define a plan, you should add a parameter set with these Plan parameters that makes the ones reuired for definiing a plan mandatory.

Reply:
Thanks.